src/platform/macos.rs

```diff
@@ -332,6 +332,8 @@ pub trait WindowAttributesExtMacOS {
     fn with_borderless_game(self, borderless_game: bool) -> Self;
     /// See [`WindowExtMacOS::set_unified_titlebar`] for details on what this means if set.
     fn with_unified_titlebar(self, unified_titlebar: bool) -> Self;
+    /// Defines kind of the window
+    fn with_window_kind(self, kind: WindowKind) -> Self;
```

Comment:
API-wise, I think I'd prefer a `with_panel(self, panel: bool) -> Self` method instead, since while there are other subclasses of `NSWindow`, namely `NSColorPanel`, `NSSavePanel`, `NSOpenPanel` and `NSFontPanel`, I don't think either of these are desirable for Winit to override.